lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -426,8 +431,8 @@ public function get($key)
     {
         if (
             ! $this->initialized
-            && $this->association['fetch'] === ClassMetadata::FETCH_EXTRA_LAZY
-            && isset($this->association['indexBy'])
+            && $this->getMapping()['fetch'] === ClassMetadata::FETCH_EXTRA_LAZY
+            && isset($this->getMapping()['indexBy'])
```

Comment:
Here you could store the result in a variable called `$association`, and use it also on line 434/439